### PR TITLE
refactor(StudentRunListComponent): Remove duplication

### PR DESCRIPTION
**Pull request by breity**
## Changes
- Combine functions to retrieve number of runs based on status.
- Extract sortByStartTimeDesc() for use in both student and teacher run lists.

## Test
- Make sure student and teacher run lists load runs properly and work as before.
- Ensure that # of runs active/running/scheduled is still calculated correctly.

### CODE REVIEW

File: src/app/domain/run.spec.ts [hunk 1]
```diff
@@ -1,14 +1,14 @@
-import { waitForAsync } from '@angular/core/testing';
-import { Run } from './run';
+import { title } from 'process';
```
Comment (hirokiterashima) [inline src/app/domain/run.spec.ts]: Unused, remove?
```suggestion
import { title } from 'process';
```

File: src/app/domain/run.spec.ts [hunk 2]
```diff
@@ -35,4 +35,21 @@ describe('Run', () => {
...
+      startTime: endTime
+    });
+    const run3 = new Run({
+      startTime: afterEndTime
+    });
+    beforeEach(() => {
+      runList = [run2, run3, run];
+      runList.sort(sortByRunStartTimeDesc);
+    });
+    it('should sort runs by start date', async () => {
```
Comment (hirokiterashima) [inline src/app/domain/run.spec.ts]: Remove async and "should"?
```suggestion
    it('sorts runs by start date', () => {
```

File: src/app/student/student-run-list/student-run-list.component.ts [hunk 3]
```diff
@@ -86,33 +79,11 @@ export class StudentRunListComponent implements OnInit {
...
-    for (let run of this.filteredRuns) {
-      if (run.isCompleted(now)) {
-        total++;
-      }
-    }
-    return total;
-  }
-
-  scheduledTotal(): number {
+  getRunTotal(type: 'isActive' | 'isScheduled'): number {
```
Comment (hirokiterashima) [inline src/app/student/student-run-list/student-run-list.component.ts]:
```suggestion
  protected getRunTotal(type: 'isActive' | 'isScheduled'): number {
```

File: src/app/student/student-run-list/student-run-list.component.ts [hunk 4]
```diff
@@ -86,33 +79,11 @@ export class StudentRunListComponent implements OnInit {
...
     let total = 0;
     const now = this.configService.getCurrentServerTime();
     for (let run of this.filteredRuns) {
```
Comment (hirokiterashima) [inline src/app/student/student-run-list/student-run-list.component.ts]:
```suggestion
    for (const run of this.filteredRuns) {
```

**Review by hirokiterashima — APPROVED** [review]
LGTM. I added some minor inline code comments to consider before merge. I also realize that we can now write getRunTotal() declaratively (e.g., ```this.filteredRuns.filter(...).length)```), which will make the code even easier to read, but we can address this at a later time if you'd like.

**Comment by breity** [comment]
@hirokiterashima Thanks, I updated with your suggestions and wrote `getRunTotal()` declaratively. Will merge.

**Pull request merged**